src/sentry/api/endpoints/organization_auth_provider.py

```diff
@@ -0,0 +1,218 @@
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models import AuditLogEntryEvent, AuthProvider, OrganizationMember
+from sentry.plugins.base.response import Response as PluginResponse
+from sentry.utils import db
+from sentry.utils.http import absolute_uri
+
+ERR_NO_SSO = _('The SSO feature is not enabled for this organization.')
+
+OK_PROVIDER_DISABLED = _('SSO authentication has been disabled.')
+
+
+class AuthSettingsSerializer(serializers.Serializer):
+    require_link = serializers.BooleanField()
+    default_role = serializers.ChoiceField(choices=roles.get_choices())
+
+
+class OrganizationAuthProviderEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationAuthProviderPermission, )
+
+    def get(self, request, organization):
+        """
+        Retrieve an Organization's Auth Provider
+        ````````````````````````````````````````
+
+        :pparam string organization_slug: the organization short name
+        :auth: required
+        """
+        if not features.has('organizations:sso', organization, actor=request.user):
+            return Response(ERR_NO_SSO, status=status.HTTP_403_FORBIDDEN)
+
+        try:
+            auth_provider = AuthProvider.objects.get(
+                organization=organization,
+            )
+        except AuthProvider.DoesNotExist:
+            # This is a valid state where org does not have an auth provider
+            # configured, make sure we respond with a 20x
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
+            # provider configure view can either be a template or a http response
+            provider = auth_provider.get_provider()
+
+            view = provider.get_configure_view()
+            response = view(request, organization, auth_provider)
+
+            if isinstance(response, HttpResponse):
+                return response
+            elif isinstance(response, PluginResponse):
+                response = response.render(
+                    request, {
+                        'auth_provider': auth_provider,
+                        'organization': organization,
+                        'provider': provider,
+                    }
+                )
+
+            pending_links_count = OrganizationMember.objects.filter(
+                organization=organization,
+                flags=~getattr(OrganizationMember.flags, 'sso:linked'),
+            ).count()
+
+            context = {
+                'pending_links_count': pending_links_count,
+                'login_url': absolute_uri(reverse('sentry-organization-home', args=[organization.slug])),
+                'auth_provider': serialize(auth_provider),
+                'default_role': organization.default_role,
+                'require_link': not auth_provider.flags.allow_unlinked,
+                'provider_name': provider.name,
+                'content': serialize(response),
```

Comment:
for whoever reviews this: the 'content' string gets HTML injected and needs to be carefully considered for XSS vulns

Reply:
here: https://github.com/getsentry/sentry/pull/6302/files#diff-ed1ee22ac547de5b38fb5706b67efcd1R108